File: 6단계/4673.c
#include <stdio.h>

int d(int num){//셀프넘버가 아닌 수 찾기
    int sum=num;
    while(num > 0){
        sum += num%10;
        num/=10;
    }
    return sum;
}

int main(void){
    int arr[10001];
    int check;
    
    for(int i=0; i<=10000; i++){//배열에 1~10000까지 넣어주고
        arr[i]=i;
    }
    for(int i=0; i<=9999; i++){//셀프넘버가 아닌 수들 0으로 초기화
        check=d(i);
        if(check<=10000){
            arr[check]=0;
        }
    }
    for(int i=0; i<=10000; i++){ //셀프넘버 출력
        if(arr[i]!=0) printf("%d\n", arr[i]);
    }
}

File: 6단계/4673.py
def not_selfnumber(num):
    a=0
    for i in list(str(num)):
        a=a+int(i)
    return int(num)+a


a=[]
for i in range(1,10001):
    k = not_selfnumber(i)
    a.append(k)


for j in range(1, 10001):
    if j in a:
        continue
    else:
        print(j)
